File: primes/primes.py

def is_prime(a):
    return all(a % i for i in range(2, int(a**.5)+1))


n_for_prime = 4
print "----------------------------------------"
for n in range(3, 1000000):
	possible_prime = n**2 - n + 41
	if is_prime(possible_prime):
		# print "The prime is " + str(possible_prime)
		# print "The value of n is: " + str(n)
		previous_n_for_prime = n_for_prime
		n_for_prime = n
		prime_gap = n - previous_n_for_prime
		if possible_prime > 1857049783:
			print "The prime is: " + str(possible_prime)
			print "The prime gap is: " + str(prime_gap)

File: primes/primes.rs
use std::num::sqrt;

fn is_prime(num: int) -> bool {
	let mut prime = true;
	for x in range(2, sqrt(num as f64) as int +1) {
		if num % x == 0 {
			prime = false;
			break
		}
	}
	return prime
}

fn main() {
	for a in range(1,40) {
		for b in range(1,40) {
			for c in range(1,40) {
				for x in range(1,40) {
					let prime = a*x*x+b*x+c;
					if is_prime(prime) {
						if x > 20 {
							println(format!("{:d}x^2 + {:d}x + {:d}, {:d}, {:d}, {:?}", a, b, c, x, prime, is_prime(prime)));
						}
					}
					else {
						break
					}
				}
			}
		}
		
	}
}